tools/ports/__init__.py

```diff
@@ -35,6 +35,10 @@
 ports_dir = os.path.dirname(os.path.abspath(__file__))
 
 logger = logging.getLogger('ports')
+logger.setLevel(logging.DEBUG)
+
+DEFAULT_BUILD_CONFIG = {'deferred': False, 'variant': ''}
+current_build_config = DEFAULT_BUILD_CONFIG
```

Comment:
Turns out we also need to separate the build directories per variant, so now we need to also pass the original variant name as well as the deferred config. This is still a global var in this file rather than having to thread it through every port's version of `get()`.
WDYT? Too ugly to land?

Reply:
Found a better way to do it.